### PR TITLE
move upsert and a handful of helpers to InMemAccountsIndex

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
pushing account index functionality into InMemAccountsIndex to prepare for disk buckets

#### Summary of Changes

Fixes #

### CODE REVIEW

File: runtime/src/accounts_index.rs [hunk 1]
```diff
@@ -2989,14 +2904,14 @@ pub mod tests {
...
-        WriteAccountMapEntry::upsert(
-            w_account_maps,
+        let mut w_account_maps = index.get_account_maps_write_lock(&key.pubkey());
+        w_account_maps.upsert(
             &key.pubkey(),
             new_entry,
             &mut SlotList::default(),
             UPSERT_PREVIOUS_SLOT_ENTRY_WAS_CACHED_FALSE,
         );
+        drop(w_account_maps);
```
Comment (brooksprumo) [inline runtime/src/accounts_index.rs:2914]: Can this be collapsed into `index.get_account_maps_write_lock(&key.pubkey()).upsert(...)`? I think that would remove the need for the `drop` too maybe?

**Review by brooksprumo — COMMENTED** [review]
Looks good, just the one comment. There's no functional changes, right, just moving the code around?

File: runtime/src/accounts_index.rs (hunk 1 above)
Comment (jeffwashington) [reply runtime/src/accounts_index.rs:2914]: yes, could be collapsed. just test code and I was trying to change as few lines as possible in these refactorings.

**Review by brooksprumo — APPROVED** [review]
lgtm

**Pull request merged**